Argo/Functions/catDecoded.swift

```diff
@@ -1,11 +1,26 @@
 public func catDecoded<T>(xs: [Decoded<T>]) -> [T] {
-  return xs.reduce([]) { accum, elem in
-    elem.map { accum + [$0] } ?? accum
+  var accum: [T] = []
+  accum.reserveCapacity(xs.count)
+  
+  for x in xs {
+    switch x {
+    case let .Success(value): accum.append(value)
+    case .Failure: continue
+    }
   }
+  
+  return accum
 }
 
 public func catDecoded<T>(xs: [String: Decoded<T>]) -> [String: T] {
-  return xs.reduce([:]) { accum, elem in
-    elem.1.map { accum + [elem.0: $0] } ?? accum
+  var accum = Dictionary<String, T>(minimumCapacity: xs.count)
+  
```

Comment:
Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)